Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -765,7 +757,16 @@ extension SwiftLanguageServer {
       let closeHelperReq = SKDRequestDictionary(sourcekitd: self.sourcekitd)
       closeHelperReq[keys.request] = self.requests.editor_close
       closeHelperReq[keys.name] = helperDocumentName
-      _ = self.sourcekitd.send(closeHelperReq, .global(qos: .utility), reply: { _ in })
+      // FIXME: (async) We might receive two concurrent document color requests for the 
+      // same document, in which race to open/close a document with the same name in 
+      // sourcekitd. The solution is to either
+      //  - Not open the helper document and instead rely on the document that is already 
+      //    open or
```

Comment:
Do you know if there's some reason we wouldn't just do this (same for the other case as well)?

Reply:
I *think* it’s because we are getting the substructure back as a reply to the open and we didn’t capture them during the initial opening/updates.

The real solution here should probably to either create a sourcekitd request that returns the document structure or migrate it to use swift-syntax. I haven’t looked into what exactly `DocumentSymbols` does and whether migrating it to swift-syntax is even possible.